### ISSUE TITLE
Improve logging for network changes detected

### ISSUE THREAD

**Issue by 20k-ultra** [balena-io/balena-supervisor#1457]
In the following code we code we log when a network change is detected but does not explain what has changed. It will be extremely beneficial to know what has changed about the network to better understand what's causing this log. This is useful for when a container is restarted due to network changes but we do not know what changes were made.

https://github.com/balena-io/balena-supervisor/blob/13241bf6e7aaa2d29b3fc9937052f2d685afc087/src/compose/service.ts#L747-L749

**Comment by pipex** [comment]
I wonder if these mistery service restarts are more common than we think. My hypothesis is that is not something that most users notice, because the info is not readily available. Probably another reason for your observability feature.

**Comment by 20k-ultra** [comment]
Yes this issue was actually the motivator for that feature @pipex :p
